shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -207,6 +208,11 @@ - (void)requestApplicationTermination:(id)sender
                              exitType:(FlutterAppExitType)type
                                result:(nullable FlutterResult)result {
   _shouldTerminate = YES;
+  if (![self bindingIsReady]) {
+    // Until the binding has signaled that it is ready to handle application
```

Comment:
Consider tweaking this as raw `dart:ui` could also opt-in to handle termination requests:

```suggestion
    // Until the Dart app has signaled that it is ready to handle application
```

Reply:
Sure, makes sense. Also renamed the variable to `acceptingRequests`.

Reply:
Sounds good!